### DISCUSSION TITLE
Cross-compilation best  practice

### DISCUSSION THREAD

**Discussion by taozuhong** [msys2/MINGW-packages#14045] [General]
AArch64、RiscV64 will be more popular in the future, we need a detailed guide to cover this, do you have suggest about it?

**Comment by Biswa96** [comment]
For AArch32 and AArch64, this project can be used https://github.com/mstorsjo/llvm-mingw with clangarm64 but with some caveats.

**Comment by taozuhong** [comment]
For now, the UCRT64 is the default environment, based on it, 
1、How to build out the AArch64、RISCV64 app？ 
2、How to pack the app for target machine?

**Reply by jeremyd2019** [reply]
MSYS2 really mostly is intended for building Windows software - through the various other environments MINGW, UCRT, CLANG....  AFAIK there is no RISCV64 version of Windows so I'm not quite sure what you're asking for there (https://packages.msys2.org/base/mingw-w64-riscv64-unknown-elf-gcc ?).  For Aarch64 Windows, we're mostly building "natively", but you can cross-compile with llvm-mingw as indicated above, or there was a discussion about cross compiling #8736 and proof-of-concept packages that work but just don't really fit in the current assumptions about packages here #8762

**Reply by taozuhong** [reply]
> what you're asking for there?

I want to write a detailed document to guide us to use UCRT64 environment to build out AArch64, RISCV64 app for windows.

I need your help, thanks in advanced.

**Reply by mati865** [reply]
> there is no RISCV64 version of Windows so I'm not quite sure what you're asking for there

**Reply by taozuhong** [reply]
AArch64 cross-compilation is ready on MSYS2?

**Reply by mati865** [reply]
As Jeremy already said there is no ready cross-compilation but there is some WIP, see his response for more info.
